### PR TITLE
[Logger] Replace logger values with pointers

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Description

Previously when setting up a new module according to the logger module readme, one encounters [`hugeParam` linter error](https://golangci-lint.run/usage/linters/#gocritic) about the module's `logger` field.

## Issue

image

## Type of change

Please mark the relevant option(s):

- [ ] New feature, functionality or library
- [ ] Bug fix
- [x] Code health or cleanup
- [ ] Major breaking change
- [ ] Documentation
- [ ] Other 

## List of changes

- Update the logger module interface to use logger pointers instead of values
- Update logger value references with pointers
- Update README

## Testing

- [x] `make develop_test`
- [x] [LocalNet](https://github.com/pokt-network/pocket/blob/main/docs/development/README.md) w/ all of the steps outlined in the `README`

## Required Checklist

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have tested my changes using the available tooling
- [x] I have updated the corresponding CHANGELOG

### If Applicable Checklist

- [x] I have updated the corresponding README(s); local and/or global
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added, or updated, [mermaid.js](https://mermaid-js.github.io) diagrams in the corresponding README(s)
- [ ] I have added, or updated, documentation and [mermaid.js](https://mermaid-js.github.io) diagrams in `shared/docs/*` if I updated `shared/*`README(s)

### CODE REVIEW

File: logger/module.go [hunk 1]
```diff
@@ -57,8 +57,13 @@ func Create(bus modules.Bus, options ...modules.ModuleOption) (modules.Module, e
 	return new(loggerModule).Create(bus, options...)
 }
 
-func (*loggerModule) CreateLoggerForModule(moduleName string) modules.Logger {
-	return Global.Logger.With().Str("module", moduleName).Logger()
+// CreateLoggerForModule returns the logger with additional context (e.g. module name)
```
Comment (Olshansk) [inline logger/module.go:60]: Not actionable.

Thanks for this. We definitely don't do a good job at using godoc at the moment, so do continue doing this whenever you find an opportunity.

File: persistence/genesis.go [hunk 2]
```diff
@@ -13,8 +13,6 @@ import (
 
 // CONSIDERATION: Should this return an error and let the caller decide if it should log a fatal error?
 func (m *persistenceModule) populateGenesisState(state *genesis.GenesisState) {
-	m.logger.Info().Msg("Populating genesis state...")
```
Comment (Olshansk) [inline persistence/genesis.go:16]: Not useful in your opinion?

File: shared/modules/logger_module.go [hunk 3]
```diff
@@ -15,7 +15,9 @@ type LoggerModule interface {
 	// TODO(#288): Embed `ObservableModule` inside of the `Module` interface so each module has its own context specific logger
 	ObservableModule
 
-	// `CreateLoggerForModule` returns the logger with additional context (e.g. module name)
-	// https://github.com/rs/zerolog#sub-loggers-let-you-chain-loggers-with-additional-context
-	CreateLoggerForModule(string) Logger
+	// CreateLoggerForModule returns the logger with additional context (e.g. module name)
```
Comment (Olshansk) [inline shared/modules/logger_module.go:18]: Do we need the same comment both next to the interface and the implementation?

Don't have a good answer to this, just feels repetative.

**Review by Olshansk — APPROVED** [review]
Thanks @bryanchriswhite! Left a couple questions/comments, but also lgtm (looks good to merge) as is!

File: shared/modules/logger_module.go (hunk 3 above)
Comment (bryanchriswhite) [reply shared/modules/logger_module.go:18]: I was similarly uncertain and defaulted to verbosity.

Comment (bryanchriswhite) [reply shared/modules/logger_module.go:18]: Wdyt about leaving the on the interface and the making the comment on the implementation something like this:
```
// CreateLoggerForModule implements the `modules.Logger` interface member.
```

**Comment by bryanchriswhite** [comment]
## CI Failure

As of 21c9fe0, tests are failing in CI due to #539. This is the only failure and can be confirmed by clicking on the gear icon --> "View raw logs" of [the respective run](https://github.com/pokt-network/pocket/actions/runs/4260978062/jobs/7414758531). Then search the raw logs for "--- fail" to find thee test failure.

File: shared/modules/logger_module.go (hunk 3 above)
Comment (Olshansk) [reply shared/modules/logger_module.go:18]: I LIKE IT!

Throw it in the code review guideline ;)

**Pull request merged**